dora/src/bytecode/data.rs

```diff
@@ -1,8 +1,42 @@
+use std::collections::HashMap;
 use std::fmt;
 
 use crate::cpu::STACK_FRAME_ALIGNMENT;
 use crate::mem::{align_i32, ptr_width};
 use crate::ty::{BuiltinType, MachineMode};
+use dora_parser::lexer::position::Position;
+
+#[derive(Debug)]
+pub struct PositionTable {
+    map: HashMap<u32, Position>,
+}
+
+impl PositionTable {
+    pub fn new() -> PositionTable {
+        PositionTable {
+            map: HashMap::new(),
+        }
+    }
+
+    pub fn insert(&mut self, offset: u32, position: Position) {
+        assert!(self.map.insert(offset, position).is_none());
+    }
+
+    pub fn get(&self, offset: u32) -> Option<&Position> {
+        self.map.get(&offset)
+    }
+
+    pub fn to_vec(&self) -> Vec<(u32, Position)> {
+        let mut as_vec = self
+            .map
+            .iter()
+            .clone()
+            .map(|(k, v)| (*k, *v))
+            .collect::<Vec<(u32, Position)>>();
+        as_vec.sort_by(|(a, _), (b, _)| a.cmp(b));
+        as_vec
+    }
+}
```

Comment:
This is ugly as hell, but because I plan to replace PositionTable as discussed, this is just temporary.

Reply:
Done.